rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,268 @@
+top-level keys in the index documents for regular bundles (`files`, `manifest`,
+`uuid`, `version` etc). Bundle tombstones are therefore compatible with
+regular bundle documents with respect to in the same index even if
+Elasticsearch's dynamic mapping is enabled.
+
+### Replication
+
+When a bundle is deleted it is removed from all replicas.
+
+### Consistency
+
+It is the administrator's responsibility to enumerate all bundles referring to
+a file that is to be deleted and issue a delete requests for each of
+them. If the administrator misses such a bundle (a *dangling bundle*), users
+requesting `GET /bundle` for a dangling bundle will be able to retrieve its
+manifest. Likewise, `POST /search` requests will return the metadata for
+dangling bundles. Secondary indexes like the Data Browser (Orange Box) will
+continue to hand out (meta)data for dangling bundles. If users request `GET
+/files` for a file contained in a dangling bundle and the requested file was
+referenced by another already logically (physically) deleted bundle, the Data
+Store will respond with a 404 (500) status code.
+
+The administrator can initiate a consistency check against a storage replica to
+list dangling bundles. The administrator can then issue more `DELETE /bundles`
+requests for those dangling bundles.
+
+### Secondary indexes
+
+The Data Store can optionally notify subscribers about the creation of a bundle
+tombstone. Secondary index services need to register a subscription using
+a query that matches deletion markers. When they are notified about a deletion
+marker, they need to take the necessary actions to remove the data from their
+database or index.
+
+### Derived data
+
+The metadata in bundles containing derived data must reference the input bundle
+versions. Administrators are responsible for tracking down data bundles with
+data derived from data in deleted bundles and to issue deletion requests for
+them.
+
+### Tiered storage
+
+Physical deletion of blobs is effective across all tiered storage classes in
+the underlying blob store. Additional cost may be incurred for deleting data recently moved to a
+slower access storage tier. Deletion occur There is no additional delays cause when deleting from a 
+tiered storage system such as AWS glacier.
+
+### Undoing logical deletions
+
+A bundle can be restored using `restore_bundle.py <uuid.<version>` if has not been physically deleted. This process
+uses the info from the deletion table to removes tombstones, and prevent permanent deletion. Subscribers of new 
+bundles will receive a notification that a new bundle has arrived.
+
+Deletion Table Entry Example:
+
+|Bundle.Version|admin|reason|AWS Deletion Markers (key,ID)|GCP Previous Generations (key, previous generation)|
+|--------------|-----|------|--------------------|------------------------|
+|1234-2134-3145|admin@email.com|consent|(file.obj, 1234), ...| (file.obj, 4321), ...|
```

Comment:
Do I understand correctly that the deletion table is an *implementation detail* of the DSS, internal to DSS and not exposed in the API? I think it's OK and appropriate to list the details of it here, as long as that fact is called out. More generally, throughout this RFC, more clarity is necessary to separate the background information, the API for the deletion actions, and the implementation details of the deletion service infrastructure.